### PR TITLE
feat(Core/SmartScripts) Implement SMART_TARGET_VEHICLE_PASSENGER

### PR DESCRIPTION
**Pull request by Nefertumm**
## CHANGES PROPOSED:
-  Implement missing target.
-  Ported from https://github.com/TrinityCore/TrinityCore/pull/19352

## ISSUES ADDRESSED:
- Related to #1141 

## TESTS PERFORMED:
- Compiled and played around with vehicles

## HOW TO TEST THE CHANGES:

Just search for a vehicle with the right smart_script target and play with it.

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -3856,6 +3856,15 @@ ObjectList* SmartScript::GetTargets(SmartScriptHolder const& e, Unit* invoker /*
         delete units;
         break;
     }
+    case SMART_TARGET_VEHICLE_PASSENGER:
+    {
+        if (me && me->IsVehicle())
+        {
+            if (Unit* target = me->GetVehicleKit()->GetPassenger(e.target.vehicle.seat))
+                l->push_back(target);
```
Comment (FrancescoBorzi) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:3864]: brackets please

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@Nefertumm why closing this?

**Comment by Si1ker** [comment]
Probably deleted branch and that auto-closed this. 🐱  but will be good to hear from him.

**Comment by FrancescoBorzi** [comment]
the branch has not been deleted: https://github.com/Nefertummm/azerothcore-wotlk/tree/vehicle_accessory

**Comment by Si1ker** [comment]
image

**Comment by BarbzYHOOL** [comment]
it was "to be merged" why not reopen it and merge it if it fixes smthg? ^^

**Comment by Nefertumm** [comment]
If anyone cares to test this, you can go to Ulduar and see vehicle's behaviour. This is where the most worldserver message spam comes from.

**Pull request merged**